hazelcast/src/test/java/com/hazelcast/config/MergePolicyConfigTest.java

```diff
@@ -0,0 +1,137 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.merge.DiscardMergePolicy;
+import com.hazelcast.spi.merge.HigherHitsMergePolicy;
+import com.hazelcast.spi.merge.PassThroughMergePolicy;
+import com.hazelcast.spi.merge.PutIfAbsentMergePolicy;
+import com.hazelcast.spi.serialization.SerializationService;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class MergePolicyConfigTest {
+
+    private MergePolicyConfig config = new MergePolicyConfig();
+
+    @Test
+    public void testConstructor_withParameters() {
+        new MergePolicyConfig(DiscardMergePolicy.class.getName(), 100);
+        new MergePolicyConfig(PassThroughMergePolicy.class.getName(), 1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructor_withParameters_withNullPolicy() {
+        new MergePolicyConfig(null, 100);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructor_withParameters_withEmptyPolicy() {
+        new MergePolicyConfig("", 100);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructor_withParameters_withZeroBatchSize() {
+        new MergePolicyConfig(DiscardMergePolicy.class.getName(), 0);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConstructor_withParameters_withNegativeBatchSize() {
+        new MergePolicyConfig(DiscardMergePolicy.class.getName(), -1);
+    }
+
+    @Test
+    public void setPolicy() {
+        config.setPolicy(DiscardMergePolicy.class.getName());
+
+        assertEquals(DiscardMergePolicy.class.getName(), config.getPolicy());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setPolicy_withNull() {
+        config.setPolicy(null);
+    }
+
+    @Test
+    public void setBatchSize() {
+        config.setBatchSize(1234);
+
+        assertEquals(1234, config.getBatchSize());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setBatchSize_withZero() {
+        config.setBatchSize(0);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void setBatchSize_withNegative() {
+        config.setBatchSize(-1);
+    }
+
+    @Test
+    public void testToString() {
+        config.setPolicy(HigherHitsMergePolicy.class.getName());
+        config.setBatchSize(2342);
+
+        String configString = config.toString();
+        assertTrue(configString.contains("MergePolicyConfig"));
```

Comment:
I like more `assertThat(configString, CoreMatchers.containsString("MergePolicyConfig"));` as it provides nice messages what went wrong.